test/ecto/query/builder_test.exs

```diff
@@ -20,6 +20,10 @@ defmodule Ecto.Query.BuilderTest do
     assert {Macro.escape(quote do &0.y > &1.z end), %{}} ==
            escape(quote do x.y > y.z end, [x: 0, y: 1], __ENV__)
 
+    import Kernel, except: [+: 2]
+    assert {Macro.escape(quote do &0.y + &1.z end), %{}} ==
```

Comment:
Let's also add a unit test that calls `type(x.y + x.z, x.y)`.

Reply:
I can add this one on the documentation PR ^^